content/blog/managing-aws-credentials-on-cicd/index.md

```diff
@@ -0,0 +1,231 @@
+- Automate rotating and revoking that user’s credentials
+- During deployments, exchange the IAM User credentials for more privileged IAM Role credentials
+
+The process may sound complicated, but we’ll cover each step in detail.  We’ll point out additional ways to enhance your security, such as using Pulumi’s support for client-side encryption to limit access to sensitive information.
+
+## Create a dedicated IAM User for your CI/CD
+
+The first step for securely automating CI/CD on AWS is to create a dedicated IAM User for use in your CI/CD pipelines.
+
+By default, a new IAM User does not have any permissions associated with it, meaning it can only authenticate itself with its access key and secret. Let’s walk through how to grant this IAM User the permissions necessary to update your cloud infrastructure later. For now, we limit this user to have as little access as possible.
+
+The following code snippet shows how to create an IAM User specifically for your CI/CD jobs. 
+
+```ts
+const aws = require("@pulumi/aws");
+
+const role = new aws.iam.Role("myrole", {
+    assumeRolePolicy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [{
+            Action: "sts:AssumeRole",
+            Principal: {
+                Service: "ec2.amazonaws.com"
+            },
+            Effect: "Allow",
+            Sid: ""
+        }]
+    })
+});
+
+const rolePolicy = new aws.iam.RolePolicy("myrolepolicy", {
+    role: role,
+    policy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [{
+            Action: [ "ec2:Describe*" ],
+            Effect: "Allow",
+            Resource: "*"
+        }]
+    })
+});
+
+const policy = new aws.iam.Policy("mypolicy", {
+    policy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [{
+            Action: [
+              "ec2:Describe*"
+            ],
+            Effect: "Allow",
+            Resource: "*"
+        }]
+    })
+});
+
+const rolePolicyAttachment = new aws.iam.RolePolicyAttachment("myrolepolicyattachment", {
+    role: role,
+    policyArn: policy.arn
+});
+
+const user = new aws.iam.User("myuser");
+
+const group = new aws.iam.Group("mygroup");
+
+const policyAttachment = new aws.iam.PolicyAttachment("mypolicyattachment", {
+    users: [user],
+    groups: [group],
+    roles: [role],
+    policyArn: policy.arn
+});
+```
+
+### Isolating with Different Accounts
+
+A common and **recommended** practice is to use multiple AWS accounts to isolate resources and users from one another. For example, if your testing and production environments are in different AWS accounts, it would be difficult for a severe bug in the testing environment to harm your customer data.
+
+Keep this in mind as we configure this IAM User and later grant them IAM Roles to perform cloud deployments. Creating this specific IAM User to CI/CD jobs with its AWS account means that there are no resources to modify, in addition to not having permissions to do anything.
```

Comment:
"jobs with its AWS account" doesn't quite read right.

Are we trying to say: "You should separate your testing and production instances in separate AWS accounts. This separation ensures that a bug in your testing environment cannot harm customer data."?

Are we trying to make an additional claim about the AWS account where the IAM User we are using for CI/CD should be created? We could make that claim, but the details are complicated and it's a pain. So I'd just mention it in a link. e.g. "It isn't a requirement that the IAM User you will use to deploy your infrastructure is in the same AWS account that the cloud resources are located in. For example, it is possible for the IAM User in your "testing" account to perform a deployment of the application housed in your "production" environment.

For the rest of this post, we'll assume that the IAM user is in the same AWS account. But for more information about IAM and cross-account access, see [the documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_common-scenarios_third-party.html). (The same approach described here, assuming an IAM Role, is used.)